Add clarification about glob chars in branch/tag matching (#8107)
github/docs: content/actions/reference/workflow-syntax-for-github-actions.md

When using the `push` and `pull_request` events, you can configure a workflow to run on specific branches or tags. For a `pull_request` event, only branches and tags on the base are evaluated. If you define only `tags` or only `branches`, the workflow won't run for events affecting the undefined Git ref.

> The `branches`, `branches-ignore`, `tags`, and `tags-ignore` keywords accept glob patterns that use characters like `*`, `**`, `+`, `?`, `!` and others to match more than one branch or tag name. If your branch name contains one of these characters, you need to *escape* the character with `\` for a literal match. For more information, see the "[Filter pattern cheat sheet](#filter-pattern-cheat-sheet)."

Reviewer: Making an update as this also affects tag names.

```suggestion
The `branches`, `branches-ignore`, `tags`, and `tags-ignore` keywords accept glob patterns that use characters like `*`, `**`, `+`, `?`, `!` and others to match more than one branch or tag name. If a name contains any of these characters and you want a literal match, you need to *escape* each of these special characters with `\`. For more information about glob patterns, see the "[Filter pattern cheat sheet](#filter-pattern-cheat-sheet)."
```

> The `branches`, `branches-ignore`, `tags`, and `tags-ignore` keywords accept glob patterns that use characters like `*`, `**`, `+`, `?`, `!` and others to match more than one branch or tag name. For more information, see the "[Filter pattern cheat sheet](#filter-pattern-cheat-sheet)." *Please remember*: if your branch name contains these characters, you'll need to *escape* them with `\` for exact matching.

Reviewer 2: I don't think adding wildcard characters is necessary when each has a detailed usage description in the [Filter pattern cheat sheet](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#filter-pattern-cheat-sheet)

Author: This has been exactly what I missed. I did not make a connection from pure mention that page to the possibility that this might derail my exact match. And there are no examples using these other chars beside *, ** and !.

At least a hint to take extra care for exact matches would be nice.

Reviewer 2: Sure!
A hint can be useful.

Reviewer: @poikilotherm - I think you're right to highlight the need to escape these characters here. It would be easy for someone to overlook the need to escape these characters, especially if they do not want to specify a glob pattern.

I think this will work a little better like this:

```suggestion
The `branches`, `branches-ignore`, `tags`, and `tags-ignore` keywords accept glob patterns that use characters like `*`, `**`, `+`, `?`, `!` and others to match more than one branch or tag name. If your branch name contains one of these characters, you need to *escape* them with `\` to for a literal match. For more information, see the "[Filter pattern cheat sheet](#filter-pattern-cheat-sheet)."
```
